### PR TITLE
Fix collection_exists endpoint validation for empty and control character names

### LINKED ISSUE #7501: collection_exists returns 404 due to wrong route instead of validating invalid names
### Description

When calling `collection_exists` with **invalid collection names** (such as empty string or control characters), the client constructs an incorrect request path like:

```
/collections//exists
```

This leads the server to return:

```
404 Not Found
{"status":{"error":"Not found: Collection `exists` doesn't exist!"}}
```

This is **not a genuine “not found” error**, but a **parameter validation issue**.
 The request should have been rejected before reaching the REST route.

------

### 🔬 Reproduction Code

```
from qdrant_client import QdrantClient

client = QdrantClient(url="http://localhost:6333")

for name in ["", "\t", "\n"]:
    try:
        client.collection_exists(collection_name=name)
    except Exception as e:
        print(f"{repr(name)} -> {type(e).__name__}: {e}")
```

------

### 🧩 Current Behavior

- The SDK/server generates `/collections//exists` for invalid names.
- The server responds with `404` and message:
   `"Collection \`exists` doesn't exist!"`

------

### ✅ Expected Behavior

- Invalid collection names (empty or control characters) should **not reach the route handler**.
- SDK or server should **validate input** and return a clear error response:
  - HTTP **400 Bad Request**, or
  - HTTP **422 Unprocessable Entity**
- The error message should explicitly state that the `collection_name` is invalid, rather than implying a missing `"exists"` collection.

------

### 💡 Suggested Fix

- **Option A (SDK level):**
   Validate `collection_name` before sending the request and raise a `ValueError` if invalid.
- **Option B (Server level):**
   Validate path parameters and return `400/422` with an appropriate error message (`"Invalid collection name"`).

------

### 🧪 Environment

| Component  | Version / Value         |
| ---------- | ----------------------- |
| Qdrant     | *v1.15.5*               |
| Python SDK | `qdrant-client==1.14.2` |
| Endpoint   | `http://localhost:6333` |

### PR DESCRIPTION
**Pull request by chen0427ok**
### All Submissions:

  * [x] Contributions should target the `dev` branch. Did you create your branch from `dev`?
  * [x] Have you followed the guidelines in our Contributing document?
  * [x] Have you checked to ensure there aren't other open Pull Requests for the same update/change?

  ### New Feature Submissions:

  1. [x] Does your submission pass tests?
  2. [x] Have you formatted your code locally using `cargo +nightly fmt --all` command prior to submission?
  3. [x] Have you checked your code using `cargo clippy --all --all-features` command?

  ### Changes to Core Features:

  * [x] Have you added an explanation of what your changes do and why you'd like us to include them?
  * [x] Have you written new tests for your core changes, as applicable?
  * [x] Have you successfully ran tests with your changes locally?

  ---

  ## Description

  Fixes #7501

  This PR fixes a validation issue in the `collection_exists` endpoint where invalid collection names (empty
  strings and control characters) return misleading 404 errors instead of proper validation errors.

  ## Problem

  When calling `collection_exists` with invalid collection names:
  - Empty string `""` → URL becomes `/collections//exists` → Returns 404 with error "Collection `exists` doesn't
   exist!"
  - Control characters (`\t`, `\n`, `\r`) → Same routing issue

  This is not a genuine "not found" error, but a parameter validation issue that should be caught before
  reaching the route handler.

  ## Solution

  Implemented **server-level validation** to ensure consistent behavior across all clients:

  ### 1. Validation Layer (`lib/common/common/src/validation.rs`)
  - Reject empty strings explicitly to prevent routing issues
  - Reject all control characters (0x00-0x1F, 0x7F) to prevent routing and display issues
  - Provide clear error messages with character codes

  ### 2. Route Handling (`src/actix/api/collections_api.rs`)
  - Add dedicated handler for `/collections/exists` edge case (when empty name is normalized)
  - Adjust route ordering to ensure more specific routes match first
  - Return HTTP 422 Unprocessable Entity with descriptive error messages

  ### 3. Integration Tests (`tests/openapi/test_collection_exists.py`)
  - Add test for empty collection name → expects 422
  - Add test for control characters → expects 422

  ## Testing

  ### Before Fix
  '' (empty)    → 404 Not Found: "Collection exists doesn't exist!"
  '\t' (tab)    → 404 Not Found: "Collection exists doesn't exist!"
  '\n' (newline) → 404 Not Found: "Collection exists doesn't exist!"

  ### After Fix
  '' (empty)    → 422 Unprocessable Entity: "collection name cannot be empty"
  '\t' (tab)    → 422 Unprocessable Entity: "control character (code: 0x09)"
  '\n' (newline) → 422 Unprocessable Entity: "control character (code: 0x0A)"
  '\r' (return)  → 422 Unprocessable Entity: "control character (code: 0x0D)"
  Valid name    → 200 OK: {"exists": false}

  ### Tests Executed
  - ✅ **Unit tests**: `cargo test -p common validate_collection_name` → Passed
  - ✅ **Clippy checks**: `cargo clippy --workspace --all-features` → No warnings
  - ✅ **Integration tests**: Manual HTTP tests with curl → All cases handled correctly
  - ✅ **Python SDK tests**: Verified error handling with qdrant-client

  ## Files Changed

  - **lib/common/common/src/validation.rs** (41 lines changed)
    - Enhanced `validate_collection_name_legacy()` with empty string check
    - Added control character validation
    - Updated unit tests

  - **src/actix/api/collections_api.rs** (16 lines changed)
    - Added `handle_empty_collection_name()` endpoint handler
    - Adjusted service configuration route ordering

  - **tests/openapi/test_collection_exists.py** (25 lines changed)
    - Added `test_empty_collection_name()` test
    - Added `test_control_character_collection_name()` test

  ## Impact

  - **Breaking change**: No - only affects invalid input that previously gave misleading errors
  - **Performance**: Negligible - validation runs before any database operations
  - **Consistency**: All clients (Python, JS, Go, Rust, .NET, Java) now receive consistent validation across 
  HTTP and gRPC interfaces
  - **Security**: Improved by preventing potential routing exploits with control characters

### CODE REVIEW

File: lib/common/common/src/validation.rs [hunk 1]
```diff
@@ -85,6 +85,28 @@ pub fn validate_collection_name(value: &str) -> Result<(), ValidationError> {
 ///
 /// This does not check the length of the name.
 pub fn validate_collection_name_legacy(value: &str) -> Result<(), ValidationError> {
```
Comment (generall) [inline lib/common/common/src/validation.rs:109]: This is a legacy validation function. We might want to change `validate_collection_name` instead

Comment (chen0427ok) [reply lib/common/common/src/validation.rs:109]: Thanks a lot for the detailed review and suggestions!

For the validate_collection_name_legacy part: the current change was originally meant to fix the concrete regression without touching validate_collection_name, but it makes sense to keep the behavior consistent between the two. I can update validate_collection_name in this PR as well and adjust the docs/comments so the intended behavior and the legacy vs non-legacy distinction are clear.

Please let me know if you have any further preferences on how strongly the tests should couple to the error messages.

**Comment by xzfc** [comment]
The server performs slash normalization. So, `GET /collections/foo/exists` is as valid as `GET ///collections/////foo////exists///`. Thus `GET /collection//exists` is not an error: it parses "get collection that named `exists`". We shouldn't change this behavior for compatibility reasons.

If a client chokes on `get_collection("")` then it should be fixed on the client side.
